### ISSUE TITLE
Please support deepseek coder greater-ly.

### ISSUE THREAD

**Issue by sprappcom** [paul-gauthier/aider#1014]
### Issue

deepseek is much cheaper and useful for certain scenario but i seemed to have hit some limits:
```
Model deepseek/deepseek-coder has hit a token limit!
Token counts below are approximate.

Input tokens: ~18,596 of 128,000
Output tokens: ~3,223 of 4,096 -- possibly exceeded output limit!
Total tokens: ~21,819 of 128,000

For more info: https://aider.chat/docs/troubleshooting/token-limits.html
```
please make it "smarter" by allowing us to break down it's output generation in piecemeals or something.

how to make this work?
```
split the router.rs file into smaller files for easier management. (do for the first 2 files first)
```
optionally i did "do the first 2 files first" when the initial prompt didnt work but it still shows token limit hit etc.

pls help suggest ways to work around. thx

### Version and model info

aider latest
ubuntu 24.04

**Comment by olafgeibig** [comment]
DeepSeek now offers a bete API with 8k output tokens. This might ease the pain. https://platform.deepseek.com/api-docs/pricing/

**Comment by sprappcom** [comment]
@olafgeibig how do i do the "beta" that offers the 8k version?

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

I'm waiting on litellm support for prefill with deepseek.

https://github.com/BerriAI/litellm/issues/4881

**Comment by mr-september** [comment]
Are the 8K tokens and the `prefill` functionality actually referring to the same thing? Or would we benefit from using both `prefill` and the 8K api endpoint instead of just one of them?

**Comment by paul-gauthier** [comment]
The main branch has 2 new features which I think solve this issue:

- Infinite output for DeepSeek Coder, Mistral models in addition to Anthropic's models.
- New `--deepseek` switch to use DeepSeek Coder.

The change is available in the main branch. You can get it by installing the latest version from github:

```
python -m pip install --upgrade git+https://github.com/paul-gauthier/aider.git
```

If you have a chance to try it, let me know if it works well for you.

**Comment by mr-september** [comment]
Is this right?
```
python -m aider.main --model deepseek --browser
```
It runs but I get this error with any prompt
```
BadRequestError: litellm.BadRequestError: GetLLMProvider Exception - list index out of range

original model: deepseek
```

**Comment by paul-gauthier** [comment]
You need to use `--model deepseek/deepseek-coder`.

I'm going to close this issue for now, but feel free to add a comment here and I will re-open or file a new issue any time.
